### PR TITLE
Fix some valgrind warnings about jumps based on uninitialized values.

### LINKED ISSUE #46905: some jumps based on uninitialized values detected by valgrind
I ran valgrind on dotnet hello world, and it pointed out a few jumps based on uninitialized values.

1. moved to https://github.com/dotnet/runtime/issues/46961

2. moved to https://github.com/dotnet/runtime/issues/46961

3. `BINDER_SPACE::AssemblyName::Init(IMDInternalImport*, _tagPEKIND, unsigned int, int) (src/coreclr/binder/assemblyname.cpp:144)`

https://github.com/dotnet/runtime/blob/36d132e7c98098d0c5566056990f6ddb0a1aee8c/src/coreclr/binder/assemblyname.cpp#L144

valgrind is not happy about `pvPublicKeyToken`, but `dwPublicKeyToken` is `0` in those cases so `pvPublicKeyToken` is not used beyond the check. Maybe `pvPublicKeyToken` can be set to `NULL` in `GetAssemblyRefProps`, or we can swap the order in the `if` to make valgrind happy?

4. moved to https://github.com/dotnet/runtime/issues/46961

5. `==6406==    at 0x563D079: PAL_VirtualUnwind (src/coreclr/pal/src/exception/seh-unwind.cpp:353)
==6406==    by 0x545253E: LazyMachState::unwindLazyState(LazyMachState*, MachState*, unsigned int, int, HostCallPreference) (src/coreclr/vm/amd64/gmsamd64.cpp:63)`

https://github.com/dotnet/runtime/blob/a1c40287acea351b0e6419cacc468d8af637f387/src/coreclr/pal/src/exception/seh-unwind.cpp#L353

`ContextFlags` is not initialized.
The context is a local at:

https://github.com/dotnet/runtime/blob/a1c40287acea351b0e6419cacc468d8af637f387/src/coreclr/vm/amd64/gmsamd64.cpp#L25

6. `==6406==    by 0x5670B1D: FreeSignalAlternateStack (src/coreclr/pal/src/thread/thread.cpp:2603)`

https://github.com/dotnet/runtime/blob/a1c40287acea351b0e6419cacc468d8af637f387/src/coreclr/pal/src/thread/thread.cpp#L2597-L2603

valgrind doesn't likes we're passing an uninitialized bytes to `sigaltstack`.
It is a false positive since that field is ignored when using SS_DISABLE.
We can probably make this warning go away by setting `ss_sp` to `NULL`.

@janvorli @jkotas can you take a look, or cc someone that can?

### PR DESCRIPTION
**Pull request by tmds**
Fixes https://github.com/dotnet/runtime/issues/46905

@janvorli ptal.

### CODE REVIEW

File: src/coreclr/vm/amd64/gmsamd64.cpp [hunk 1]
```diff
@@ -22,7 +22,7 @@ void LazyMachState::unwindLazyState(LazyMachState* baseState,
     }
     CONTRACTL_END;
 
-    CONTEXT                         ctx;
+    CONTEXT                         ctx = { 0 };
```
Comment (tmds) [inline src/coreclr/vm/amd64/gmsamd64.cpp:25]: This is a real one. It leads to uninitialized use of `ContextFlags` here:

https://github.com/dotnet/runtime/blob/a1c40287acea351b0e6419cacc468d8af637f387/src/coreclr/pal/src/exception/seh-unwind.cpp#L353

File: src/coreclr/binder/assemblyname.cpp [hunk 2]
```diff
@@ -141,7 +141,7 @@ namespace BINDER_SPACE
         }
 
         // Set public key and/or public key token (if we have it)
-        if (pvPublicKeyToken && dwPublicKeyToken)
+        if (dwPublicKeyToken && pvPublicKeyToken)
```
Comment (tmds) [inline src/coreclr/binder/assemblyname.cpp:144]: `pvPublicKeyToken` may be uninitialized, but then `pvPublicKeyToken` is `0` so there is no real issue here.
Swapping the order to make valgrind happy.

File: src/coreclr/pal/src/thread/thread.cpp [hunk 3]
```diff
@@ -2600,6 +2600,7 @@ CPalThread::FreeSignalAlternateStack()
         // ss_size is >= MINSIGSTKSZ even in this case.
         ss.ss_size = MINSIGSTKSZ;
         ss.ss_flags = SS_DISABLE;
+        ss.ss_sp = NULL;
```
Comment (tmds) [inline src/coreclr/pal/src/thread/thread.cpp:2603]: `ss_sp` is not used when `SS_DISABLE` so there is no real issue here.
Initializing to zero to make valgrind happy.

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

File: src/coreclr/vm/amd64/gmsamd64.cpp (hunk 1 above)
Comment (janvorli) [reply src/coreclr/vm/amd64/gmsamd64.cpp:25]: Good catch!

**Review by stephentoub — APPROVED** [review]

File: src/coreclr/vm/amd64/gmsamd64.cpp (hunk 1 above)
Comment (jkotas) [reply src/coreclr/vm/amd64/gmsamd64.cpp:25]: We have the same code for arm64 and arm. Does it need to be fixed there as well?

Also, `CONTEXT` is a pretty big structure. Would it better to just set the `ContextFlags` that are causing the problem?

Comment (janvorli) [reply src/coreclr/vm/amd64/gmsamd64.cpp:25]: I agree with both comments, I am sorry for not realizing it before.

**Pull request merged**